src/client/dfs/dfs.c

```diff
@@ -4346,6 +4378,17 @@ dfs_move(dfs_t *dfs, dfs_obj_t *parent, char *name, dfs_obj_t *new_parent,
 	return rc;
 }
 
+/* Wrapper function, only permit oid as a input parameter and return if it has data */
+int
+dfs_move(dfs_t *dfs, dfs_obj_t *parent, char *name, dfs_obj_t *new_parent,
+	 char *new_name, daos_obj_id_t *oid)
+{
+	if (oid && (oid->lo || oid->hi))
```

Comment:
same issue as remove. this check breaks bwd compatibility IMO and should not be here. a dfs_move_internal + flag should handle the 2 variations of the function.